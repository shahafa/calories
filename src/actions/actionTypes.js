export const RESET_AUTH_STORE = 'RESET_AUTH_STORE';
export const LOGIN_REQUEST = 'LOGIN_REQUEST';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_FAILURE = 'LOGIN_FAILURE';
export const SIGNUP_REQUEST = 'SIGNUP_REQUEST';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';

export const GET_MEALS_REQUEST = 'GET_MEALS_REQUEST';
export const GET_MEALS_SUCCESS = 'GET_MEALS_SUCCESS';
export const GET_MEALS_FAILURE = 'GET_MEALS_FAILURE';
export const ADD_MEAL_REQUEST = 'ADD_MEAL_REQUEST';
export const ADD_MEAL_SUCCESS = 'ADD_MEAL_SUCCESS';
export const ADD_MEAL_FAILURE = 'ADD_MEAL_FAILURE';
export const DELETE_MEAL_REQUEST = 'DELETE_MEAL_REQUEST';
export const DELETE_MEAL_SUCCESS = 'DELETE_MEAL_SUCCESS';
export const DELETE_MEAL_FAILURE = 'DELETE_MEAL_FAILURE';
export const EDIT_MEAL_REQUEST = 'EDIT_MEAL_REQUEST';
export const EDIT_MEAL_SUCCESS = 'EDIT_MEAL_SUCCESS';
export const EDIT_MEAL_FAILURE = 'EDIT_MEAL_FAILURE';
export const CLOSE_MEALS_ERROR_SNACKBAR = 'CLOSE_MEALS_ERROR_SNACKBAR';
export const SET_FILTER = 'SET_FILTER';

export const GET_USERS_REQUEST = 'GET_USERS_REQUEST';
export const GET_USERS_SUCCESS = 'GET_USERS_SUCCESS';
export const GET_USERS_FAILURE = 'GET_USERS_FAILURE';
export const UPDATE_USERS_ROLE_REQUEST = 'UPDATE_USERS_ROLE_REQUEST';
export const UPDATE_USERS_ROLE_SUCCESS = 'UPDATE_USERS_ROLE_SUCCESS';
export const UPDATE_USERS_ROLE_FAILURE = 'UPDATE_USERS_ROLE_FAILURE';

export const GET_SETTINGS_REQUEST = 'GET_SETTINGS_REQUEST';
export const GET_SETTINGS_SUCCESS = 'GET_SETTINGS_SUCCESS';
export const GET_SETTINGS_FAILURE = 'GET_SETTINGS_FAILURE';
export const SET_SETTINGS_REQUEST = 'SET_SETTINGS_REQUEST';
export const SET_SETTINGS_SUCCESS = 'SET_SETTINGS_SUCCESS';
export const SET_SETTINGS_FAILURE = 'SET_SETTINGS_FAILURE';
export const CLOSE_SETTINGS_ERROR_SNACKBAR = 'CLOSE_SETTINGS_ERROR_SNACKBAR';
