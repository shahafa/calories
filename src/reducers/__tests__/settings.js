import settingsReducer from '../settings';
import * as types from '../../constants/actionTypes';

describe('settings reducer', () => {
  it('should return the initial state', () => {
    expect(
      settingsReducer(undefined, {}),
    ).toMatchSnapshot();
  });

  it('should start get settings request', () => {
    expect(
      settingsReducer(undefined, {
        type: types.GET_SETTINGS_REQUEST,
      }),
    ).toMatchSnapshot();
  });

  it('should set settings after successful get settings request', () => {
    expect(
      settingsReducer(undefined, {
        type: types.GET_SETTINGS_SUCCESS,
        payload: {
          data: {
            numberOfCaloriesPerDay: 35,
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should set seetings loading flag to false after get settings failure', () => {
    expect(
      settingsReducer(undefined, {
        type: types.GET_SETTINGS_FAILURE,
      }),
    ).toMatchSnapshot();
  });

  it('should update settings locally before update settings request', () => {
    expect(
      settingsReducer(undefined, {
        type: types.UPDATE_SETTINGS_REQUEST,
        meta: {
          settings: {
            numberOfCaloriesPerDay: 35,
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should update settings after successful update settings request', () => {
    expect(
      settingsReducer(undefined, {
        type: types.UPDATE_SETTINGS_SUCCESS,
        payload: {
          data: {
            numberOfCaloriesPerDay: 35,
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should revert settings after update settings failure', () => {
    expect(
      settingsReducer(undefined, {
        type: types.UPDATE_SETTINGS_FAILURE,
      }),
    ).toMatchSnapshot();
  });
});
