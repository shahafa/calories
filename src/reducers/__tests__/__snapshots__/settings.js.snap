// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings reducer should return the initial state 1`] = `
Object {
  "isLoading": false,
  "settings": null,
}
`;

exports[`settings reducer should revert settings after update settings failure 1`] = `
Object {
  "isLoading": false,
  "settings": null,
}
`;

exports[`settings reducer should set seetings loading flag to false after get settings failure 1`] = `
Object {
  "isLoading": false,
  "settings": Object {
    "numberOfCaloriesPerDay": 35,
  },
}
`;

exports[`settings reducer should set settings after successful get settings request 1`] = `
Object {
  "isLoading": false,
  "settings": Object {
    "numberOfCaloriesPerDay": 35,
  },
}
`;

exports[`settings reducer should start get settings request 1`] = `
Object {
  "isLoading": true,
  "settings": null,
}
`;

exports[`settings reducer should update settings after successful update settings request 1`] = `
Object {
  "isLoading": false,
  "settings": null,
}
`;

exports[`settings reducer should update settings locally before update settings request 1`] = `
Object {
  "isLoading": false,
  "settings": null,
}
`;
